processor/src/chiplets/aux_trace/mod.rs

```diff
@@ -971,7 +956,7 @@ where
 /// of alphas of matching length. This can be used to build the value for a single word or for an
 /// entire [HasherState].
 fn build_value<E: FieldElement<BaseField = Felt>>(alphas: &[E], elements: &[Felt]) -> E {
-    assert_eq!(alphas.len(), elements.len());
+    debug_assert_eq!(alphas.len(), elements.len());
     let mut value = E::ZERO;
     for (&alpha, &element) in alphas.iter().zip(elements.iter()) {
         value += alpha.mul_base(element);
```

Comment:
I would add `#[inline(always)]` here.

Also, I wonder if the compiler would be able to unroll the loops here. Since we always know the length of `elements` at compile time, we could change this to look like so:
```rust
fn build_value<E: FieldElement<BaseField = Felt>, const N: usize>(alphas: &[E], elements: &[Felt; N]) -> E {
```
And then use `N` in the loop condition. Not sure if this will work, but at least there is enough info for the compiler to try to do loop unrolling.

Reply:
Changed it to your suggestion, except with `elements: [Felt; N]` (with the use of `N` explicitly in the loop just in case). It should indeed be enough for the compiler to unroll.